scipy/sparse/linalg/_dsolve/_superluobject.c

```diff
@@ -721,6 +871,26 @@ PyObject *newSuperLUObject(SuperMatrix * A, PyObject * option_dict,
     self->py_csc_construct_func = NULL;
     self->type = intype;
 
+    /* Call the appropriate function to compute the 1-norm of the matrix */
+    char norm_c = '1';  /* default to 1-norm */
+    switch (self->type) {
+        case NPY_FLOAT:
```

Comment:
It is better to spell out the bit width for more explicit dtypes in case some architecture changes these, that is NPY_FLOAT64, NPY_COMPLEX128 etc. even though they are typically aliases.